main.js

```diff
@@ -0,0 +1,123 @@
+"use script";
+var counters = {};
+var arr = [];
+var btnSort = document.getElementById('sort'); 
+var startContainer = document.getElementById('start-container');
+var endContainer = document.getElementById('end-container');
+var blockRes = document.getElementById('res');
+var ul = document.createElement('ul');
+var btn = document.getElementById('btn');
+
+//Функция для формирования массива с фильтрованными данными. 
+function getArr(){
+    var str = btnSort.value;
+    if(!str){alert('Заполните поле!'); return;}
+    str = str.replace(/ /g, "")
+    //Убираем с конца запятую если есть.
+    if(str.slice(-1) == ','){
+        str = str.slice(0, -1);
+    }
```

Comment:
May be. I used to write a single line scan and return "if (check) return;" for brevity code. Do you think this is bad? If so, I will not do that.